### ISSUE TITLE
Reporting & analytics V1

### ISSUE THREAD

**Issue by tnolet** [checkly/public-roadmap#66]
# Problem Statement
"As an engineer I want to have a longer term view on how my checks have performed over time for reporting to stakeholders in my company"

# Proposed Work
In V1, we are creating an API endpoint for generating reports:
-  per day, per week and per month reports on checks and/or checks groups.
- group by tag
- group by check group
- filter by tag
- filter by check group
- exporting reports as CSV or JSON. 
- reports success ratios, response times (avg/p95/p99)
- report aggregated metrics over total period and per time unit

Stretch goal or V2:
Graphical interface for reporting and downloading CSV

**Comment by tnolet** [comment]
Small update. We are moving ahead with this project as we speak. There are 3 parts to this:

1. optimize our data pipeline. We've seen some excellent growth and we need to replace/refactor the parts that serve the interesting statistics. This will open up the road to...
2. provide a reporting API, as described above.
3. provide a UI on top of that API.

Work is underway for part 1 of this story

**Comment by tnolet** [comment]
https://github.com/checkly/public-roadmap/issues/57

**Comment by tnolet** [comment]
Happy to report we are very close to shipping a beta of our reporting feature. We are ironing out some minor kinks. This should land pretty soon. Here is teaser screenshot from our own live production account.

![image](https://user-images.githubusercontent.com/3802923/108564811-fd7f3300-7303-11eb-9e78-54faaf0e6331.png)

**Comment by tnolet** [comment]
Hi all, 

we just shipped the V1 for reporting. Check it out at https://app.checklyhq.com/reporting
The spec changed a little from the description in the top comment. We shipped:

In V1, we are creating an API endpoint for generating reports:
- per day, per week and per month reports on checks and/or checks groups.
- filter by tag, check type.
- exporting reports as CSV. 
- reports success ratios, response times (avg/p95/p99)
- report aggregated metrics over total period and per time unit

The other features mentioned (e.g. group by tag etc.) will be picked up as we iterate and collect user feedback.
